### PR TITLE
[fix #280] fix cdc stuck when send SIGSTOP to pd leader

### LINKED ISSUE #280: cdc: `WithRequireLeader` is useless for `Campaign` stuck
### 1. Describe the bug
In capture.go#[Line387](https://github.com/tikv/migration/blob/main/cdc/cdc/capture/capture.go#L387),
`WithRequireLeader` is useless for `Campaign` stuck

image

### 2. Minimal reproduce step (Required)
1. startup 3 pd and 2 cdc
2. kill pd leader and watch cdc log
3. cdc will block `Campaign`

### 3. What did you see instead (Required)
 `Campaign` can't return
### 4. What did you expect to see? (Required)
 `Campaign` return
### 5. What is your migration tool and TiKV version? (Required)

- TiKV:
- TiKV CDC: master
- TiKV BR:
- TiKV Online Bulk Load:

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@pingcap.com>

### What problem does this PR solve?

Issue Number: close #280

Problem Description: **TBD**

### What is changed and how does it work?
Create a separate `etcd client` for `capture`. When `capture` restart, the `etcd client` will be recreated.

### Code changes

- Has exported function/method change
- Has exported variable/fields change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Side effects

- No side effects

### Related changes

- No related changes

### CODE REVIEW

**Comment by zeminzhou** [comment]
PTAL~

File: cdc/cdc/capture/capture.go [hunk 1]
```diff
@@ -434,6 +434,27 @@ func (c *Capture) AsyncClose() {
...
+
+	if c.etcdClient != nil {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.election.Resign(timeoutCtx); err != nil {
+			log.Warn("failed to resign", zap.Error(err))
+		}
+		cancel()
+
+		timeoutCtx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.etcdClient.DeleteCaptureInfo(timeoutCtx, c.info.ID); err != nil {
```
Comment (haojinming) [inline cdc/cdc/capture/capture.go:448]: Why is `DeleteCaptureInfo` moved here?

File: cdc/cdc/capture/capture.go [hunk 2]
```diff
@@ -434,6 +434,27 @@ func (c *Capture) AsyncClose() {
 		c.regionCache.Close()
 		c.regionCache = nil
 	}
+
+	c.cancel()
+
+	if c.etcdClient != nil {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.election.Resign(timeoutCtx); err != nil {
```
Comment (pingyu) [inline cdc/cdc/capture/capture.go:442]: There is another `Resign` in `campaignOwner`. Why we call again here ?

File: cdc/cdc/capture/capture.go (hunk 1 above)
Comment (pingyu) [reply cdc/cdc/capture/capture.go:448]: It seems because of the `c.etcdClient = nil` at the end of this function, where is before the original point.

Comment (zeminzhou) [reply cdc/cdc/capture/capture.go:448]: yesh!

File: cdc/cdc/capture/capture.go (hunk 2 above)
Comment (zeminzhou) [reply cdc/cdc/capture/capture.go:442]: When exiting normally, `cancel` will be called, later `owner routine` & `processor routine` exit  normally. For a `non-owner` 's `owner routine`, it should return from `campagin`.

In etcd `campagin`, when `cancel` is called, `owner routine` will return from `waitDelete`[g](https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/election.go#L93) and call `resign`[g](https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/election.go#L98)`, but now ctx is `cancel`, so `resign` will not work.

So we should call it agait.

Comment (zeminzhou) [reply cdc/cdc/capture/capture.go:442]: From detail, https://github.com/pingcap/tiflow/pull/6284

**Review by haojinming — APPROVED** [review]
LGTM

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**